internal/controller/storageclaim_controller.go

```diff
@@ -497,6 +498,10 @@ func (r *StorageClaimReconciler) getCephRBDStorageClass(data map[string]string)
 		Provisioner:          csi.GetRBDDriverName(),
 		Parameters:           data,
 	}
+
+	if r.storageClaim.Spec.EncryptionMethod != "" {
+		utils.AddAnnotation(storageClass, keyRotationAnnotation, utils.CronScheduleWeekly)
+	}
```

Comment:
I might be missing something, but it seems that this annotation is already present at line 493. What is the point of adding it again here? if the intent was to add a condition then it should not be present on line 493 (which I am not sure is the intent). They were the code is written right now, the condition itself is irrelevant and the annotation will always be present on the returned storage class.

There are 2 options here:
1. If having the annotation on the storageclass has no side-effects in cases where encryption is not enabled then these lines need to be removed.
2 If, on the other hand, having the annotation always set has side effects in non-encrypted cases, then you should delete line 493

Reply:
The annotation on line 493 pertains to `reclaim space` while the current one refers to `key rotation`.

Reply:
Checking...